tests/framework/extensions/provisioning/creates.go

```diff
@@ -227,22 +227,30 @@ func CreateProvisioningCustomCluster(client *rancher.Client, externalNodeProvide
 		}
 	}
 
-	if clustersConfig.Hardened && clustersConfig.KubernetesVersion <= string(provisioninginput.HardenedKubeVersion) {
-		err = hardening.HardeningNodes(client, clustersConfig.Hardened, nodes, rolesPerNode)
-		if err != nil {
-			return nil, err
-		}
+	if clustersConfig.Hardened {
+		if strings.Contains(clustersConfig.KubernetesVersion, "k3s") {
+			err = k3sHardening.HardeningNodes(client, clustersConfig.Hardened, nodes, rolesPerNode, clustersConfig.KubernetesVersion)
+			if err != nil {
+				return nil, err
+			}
 
-		hardenCluster := clusters.HardenK3SRKE2ClusterConfig(clusterName, namespace, clustersConfig, nil, "")
+			hardenCluster := clusters.HardenK3SClusterConfig(clusterName, namespace, clustersConfig, nil, "")
+			_, err := clusters.UpdateK3SRKE2Cluster(client, clusterResp, hardenCluster)
+			if err != nil {
+				return nil, err
+			}
 
-		_, err := clusters.UpdateK3SRKE2Cluster(client, clusterResp, hardenCluster)
-		if err != nil {
-			return nil, err
-		}
+		} else {
+			err = rke2Hardening.HardeningNodes(client, clustersConfig.Hardened, nodes, rolesPerNode)
+			if err != nil {
+				return nil, err
+			}
 
-		err = hardening.PostHardeningConfig(client, clustersConfig.Hardened, nodes, rolesPerNode)
-		if err != nil {
-			return nil, err
+			hardenCluster := clusters.HardenRKE2ClusterConfig(clusterName, namespace, clustersConfig, nil, "")
+			_, err := clusters.UpdateK3SRKE2Cluster(client, clusterResp, hardenCluster)
+			if err != nil {
+				return nil, err
+			}
```

Comment:
same nit here, this could be moved down a conditional level since it is shared among the if/else statement at this level.